2009-10-19  Jakub Jelinek  <jakub@redhat.com>

	* config/rs6000/rs6000.c (def_builtin): Set TREE_READONLY instead
	of TREE_CONSTANT.

--- gcc/config/rs6000/rs6000.c	2009-10-15 23:23:25.000000000 +0200
+++ gcc/config/rs6000/rs6000.c	2009-10-19 23:51:26.000000000 +0200
@@ -8530,7 +8530,7 @@ def_builtin (int mask, const char *name,
 
 	  /* const function, function only depends on the inputs.  */
 	case RS6000_BTC_CONST:
-	  TREE_CONSTANT (t) = 1;
+	  TREE_READONLY (t) = 1;
 	  TREE_NOTHROW (t) = 1;
 	  break;
 
@@ -8554,7 +8554,7 @@ def_builtin (int mask, const char *name,
 	      DECL_IS_NOVOPS (t) = 1;
 	    }
 	  else
-	    TREE_CONSTANT (t) = 1;
+	    TREE_READONLY (t) = 1;
 	  break;
 	}
     }
